api/resources_portal/test/views/test_organization_grant.py

```diff
@@ -0,0 +1,174 @@
+    grant_json["organizations"] = organization_list
+
+    material_list = []
+    for material in grant_json["materials"]:
+        material_list.append(material.id)
+    grant_json["materials"] = material_list
+
+    return grant_json
+
+
+class OrganizationGrantTestCase(APITestCase):
+    """
+    Tests /organizations/<id>/grants operations.
+    """
+
+    def setUp(self):
+        self.grant = GrantFactory()
+        self.grant2 = GrantFactory()
+
+        self.organization1 = self.grant.organizations.first()
+        self.org_1_member = UserFactory()
+        self.organization1.members.add(self.org_1_member)
+        self.organization1.grants.add(self.grant2)
+        self.organization1.save()
+
+        self.grant.users.add(self.organization1.owner)
+        self.grant.save()
+
+        self.organization2 = self.grant.organizations.last()
+        self.url = reverse("organization-detail", args=[self.grant.id])
+
+    def test_get_single_organization_not_allowed(self):
+        self.client.force_authenticate(user=self.organization1.owner)
+        url = reverse("organizations-grants-detail", args=[self.organization1.id, self.grant.id])
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 405)
+
+    def test_get_request_from_owner_returns_grants(self):
+        self.client.force_authenticate(user=self.organization1.owner)
+        url = reverse("organizations-grants-list", args=[self.organization1.id])
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.json()["count"], 2)
+
+    def test_get_request_from_member_returns_grants(self):
+        self.client.force_authenticate(user=self.org_1_member)
+        url = reverse("organizations-grants-list", args=[self.organization1.id])
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.json()["count"], 2)
+
+    def test_get_fails_if_not_member_or_owner(self):
+        self.client.force_authenticate(user=self.organization1.owner)
+        url = reverse("organizations-grants-list", args=[self.organization2.id])
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 403)
+
+    def test_get_fails_if_not_authenticated(self):
+        self.client.force_authenticate(user=None)
+        url = reverse("organizations-grants-list", args=[self.organization1.id])
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 403)
+
+    def test_post_request_associates_a_grant(self):
+        self.client.force_authenticate(user=self.organization1.owner)
+
+        grant = GrantFactory()
+        grant.users.add(self.organization1.owner)
+        grant.save()
+
+        url = reverse("organizations-grants-list", args=[self.organization1.id])
+        grant_json = transform_grant_json(model_to_dict(grant))
```

Comment:
Why is this call necessary?

Reply:
Call was removed, replaced with GET as discussed.